From ce08e6bf8fc288c98285e934f9532248f225f7f2 Mon Sep 17 00:00:00 2001
From: Alexandre Rostovtsev <arostovtsev@google.com>
Date: Tue, 8 Aug 2023 16:08:50 -0400
Subject: [PATCH] Require python3 for executable scripts

Otherwise, on old Linux systems where python == python2, we try to use python2
and fail: https://buildkite.com/bazel/bazel-skylib/builds/2768#0189d6bc-463d-4c32-ad02-46ed3c01da5d
---
 pkg/make_rpm.py                        | 2 ++
 pkg/private/deb/make_deb.py            | 2 ++
 pkg/private/tar/build_tar.py           | 2 ++
 pkg/private/zip/build_zip.py           | 2 ++
 pkg/releasing/git_changelog_private.py | 2 ++
 pkg/releasing/print_rel_notes.py       | 2 ++
 pkg/releasing/release_tools_test.py    | 2 ++
 pkg/verify_archive_test_main.py.tpl    | 2 ++
 8 files changed, 16 insertions(+)

diff --git a/pkg/make_rpm.py b/pkg/make_rpm.py
index e2ffca0..5c99ef2 100644
--- a/pkg/make_rpm.py
+++ b/pkg/make_rpm.py
@@ -1,3 +1,5 @@
+#!/usr/bin/env python3
+
 # Copyright 2017 The Bazel Authors. All rights reserved.
 #
 # Licensed under the Apache License, Version 2.0 (the "License");
diff --git a/pkg/private/deb/make_deb.py b/pkg/private/deb/make_deb.py
index f5efa9c..6ff3190 100644
--- a/pkg/private/deb/make_deb.py
+++ b/pkg/private/deb/make_deb.py
@@ -1,3 +1,5 @@
+#!/usr/bin/env python3
+
 # Copyright 2015 The Bazel Authors. All rights reserved.
 #
 # Licensed under the Apache License, Version 2.0 (the "License");
diff --git a/pkg/private/tar/build_tar.py b/pkg/private/tar/build_tar.py
index fde98d2..a8c4914 100644
--- a/pkg/private/tar/build_tar.py
+++ b/pkg/private/tar/build_tar.py
@@ -1,3 +1,5 @@
+#!/usr/bin/env python3
+
 # Copyright 2015 The Bazel Authors. All rights reserved.
 #
 # Licensed under the Apache License, Version 2.0 (the "License");
diff --git a/pkg/private/zip/build_zip.py b/pkg/private/zip/build_zip.py
index fe8ffab..dd3c67e 100644
--- a/pkg/private/zip/build_zip.py
+++ b/pkg/private/zip/build_zip.py
@@ -1,3 +1,5 @@
+#!/usr/bin/env python3
+#
 # Copyright 2015 The Bazel Authors. All rights reserved.
 #
 # Licensed under the Apache License, Version 2.0 (the "License");
diff --git a/pkg/releasing/git_changelog_private.py b/pkg/releasing/git_changelog_private.py
index 8344cef..40a4295 100644
--- a/pkg/releasing/git_changelog_private.py
+++ b/pkg/releasing/git_changelog_private.py
@@ -1,3 +1,5 @@
+#!/usr/bin/env python3
+
 # Copyright 2021 The Bazel Authors. All rights reserved.
 #
 # Licensed under the Apache License, Version 2.0 (the "License");
diff --git a/pkg/releasing/print_rel_notes.py b/pkg/releasing/print_rel_notes.py
index 379d0c0..5afc854 100644
--- a/pkg/releasing/print_rel_notes.py
+++ b/pkg/releasing/print_rel_notes.py
@@ -1,3 +1,5 @@
+#!/usr/bin/env python3
+
 # Copyright 2019 The Bazel Authors. All rights reserved.
 #
 # Licensed under the Apache License, Version 2.0 (the "License");
diff --git a/pkg/releasing/release_tools_test.py b/pkg/releasing/release_tools_test.py
index 7ca80d1..0a538db 100644
--- a/pkg/releasing/release_tools_test.py
+++ b/pkg/releasing/release_tools_test.py
@@ -1,3 +1,5 @@
+#!/usr/bin/env python3
+
 # Copyright 2019 The Bazel Authors. All rights reserved.
 #
 # Licensed under the Apache License, Version 2.0 (the "License");
diff --git a/pkg/verify_archive_test_main.py.tpl b/pkg/verify_archive_test_main.py.tpl
index 2d6a192..2ece337 100644
--- a/pkg/verify_archive_test_main.py.tpl
+++ b/pkg/verify_archive_test_main.py.tpl
@@ -1,3 +1,5 @@
+#!/usr/bin/env python3
+
 # Copyright 2023 The Bazel Authors. All rights reserved.
 #
 # Licensed under the Apache License, Version 2.0 (the "License");
-- 
2.41.0.640.ga95def55d0-goog

